ui/ui-components/client/client.ts

```diff
@@ -83,92 +89,90 @@ export class Client {
   private serverToken = process.env.SERVER_TOKEN;
   private cache = new ClientCache();
 
-  private static readonly optionDefaults: ClientRequestOptions = {
+  private static readonly defaultOptions: ClientRequestOptions = {
     useCache: true,
-    errorHandler,
+    errorHandler: isBrowser() ? errorHandler : undefined,
```

Comment:
The browser side seems better fit to use the `errorHandler`. The rest of the pages can use the `hydrationError` component.